### PR TITLE
Fix the QuantizedAVX2 build issue

### PR DESCRIPTION
**Pull request by lly-zero-one**
The QuantizedAVx2 does not support the int32 type. We switch to use at::quantize_vec function instead.

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 1]
```diff
@@ -395,8 +395,12 @@ void do_avg_pool_on_AVX2(
       float acc_fp[vec_width];
       acc.store(acc_int);
       vec256::convert(acc_int, acc_fp, vec_width);
-      vec256::QuantizeAvx2<T>(
-          acc_fp, o_p + c, vec_width, multiplier, output_zero_point);
+      at::quantize_vec<T>(
+          1.0f / multiplier,
+          output_zero_point,
+          acc_fp,
+          reinterpret_cast<T*>(o_p + c),
```
Comment (llyfacebook) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp]: This seems to be incorrect.

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 2]
```diff
@@ -395,8 +395,12 @@ void do_avg_pool_on_AVX2(
       float acc_fp[vec_width];
       acc.store(acc_int);
       vec256::convert(acc_int, acc_fp, vec_width);
-      vec256::QuantizeAvx2<T>(
-          acc_fp, o_p + c, vec_width, multiplier, output_zero_point);
+      at::quantize_vec<T>(
```
Comment (llyfacebook) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp]: Type seems to be incorrect.

**Comment by jamesr66a** [comment]
Any perf checks?

**Comment by llyfacebook** [comment]
> Any perf checks?

Yes,  I found it is even faster than the original one.  (For bilinear2d interpolate case.)

```
**** torch.qint8 *****
/home/lingyiliu/local/anaconda3/lib/python3.7/site-packages/torch/nn/functional.py:2494: UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
time/iter ms (float)    time/iter ms (quant)    quant/float
2.776181697845459       0.03426074981689453     0.012340960911702442
GB/s float      GB/s quant
1.165940976598206       23.619331285066114
**** torch.quint8 *****
time/iter ms (float)    time/iter ms (quant)    quant/float
2.8192615509033203      0.032978057861328125    0.011697409859245451
GB/s float      GB/s quant
1.1481247630121 24.538012620474262
**** torch.qint32 *****
time/iter ms (float)    time/iter ms (quant)    quant/float
2.0195364952087402      1.0663747787475586      0.5280294667996766
GB/s float      GB/s quant
1.602775690203824       3.0353906192358084
```

**Comment by hx89** [comment]
I got some undeclared identifier errors when build locally:
stderr: caffe2/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:654:523: error: use of undeclared identifier 'area_pixel_compute_s
cale'
caffe2/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:654:553: error: expected '(' for function-style cast or type construction
caffe2/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:654:621: error: use of undeclared identifier 'area_pixel_compute_scale'

**Comment by jamesr66a** [comment]
@llyfacebook yeah, QuantizeAVX2 operates on vectors of 32, but you were feeding it vectors of 8, so it was just running scalar code :p

**Comment by jamesr66a** [comment]
@llyfacebook I'd expect another 2x speedup if you switch to doing the float operations 4-wide and using QuantizeAVX2 again

**Review by hx89 — APPROVED** [review]
LGTM! My local build pass after rebasing.

**Pull request closed without merging**